apps/test/unit/p5lab/AnimationPicker/AnimationUploadButtonTest.js

```diff
@@ -7,84 +7,121 @@ import AnimationPickerListItem from '../../../../src/p5lab/AnimationPicker/Anima
+        inRestrictedShareMode: true
+      };
+
+      const body = shallow(<AnimationUploadButton {...combinedProps} />);
+      const uploadButton = body.find(AnimationPickerListItem).at(0);
+      uploadButton.simulate('click');
+      const warningModal = body.find(BaseDialog);
+      expect(warningModal.at(0).props().isOpen).to.be.false;
+    });
+
+    it('does not show warning if should not restrict', () => {
+      const combinedProps = {
+        ...defaultProps,
+        shouldWarnOnAnimationUpload: false
+      };
+
+      const body = shallow(<AnimationUploadButton {...combinedProps} />);
+      const uploadButton = body.find(AnimationPickerListItem).at(0);
+      uploadButton.simulate('click');
+      const warningModal = body.find(BaseDialog);
+      expect(warningModal.at(0).props().isOpen).to.be.false;
+    });
 
-  it('does not show warning if should not restrict', () => {
-    const body = shallow(
-      <AnimationUploadButton
-        onUploadClick={emptyFunction}
-        shouldRestrictAnimationUpload={false}
-        isBackgroundsTab={false}
-        refreshInRestrictedShareMode={emptyFunction}
-        inRestrictedShareMode={false}
-        showingUploadWarning={emptyFunction}
-        exitedUploadWarning={emptyFunction}
-      />
-    );
-    const uploadButton = body.find(AnimationPickerListItem).at(0);
-    uploadButton.simulate('click');
-    const warningModal = body.find(BaseDialog);
-    expect(warningModal.at(0).props().isOpen).to.be.false;
+    it('warning message requires both checkboxes to be checked to go forward', () => {
+      const body = shallow(<AnimationUploadButton {...defaultProps} />);
+      const uploadButton = body.find(AnimationPickerListItem).at(0);
+      uploadButton.simulate('click');
+      const warningModal = body.find(BaseDialog).at(0);
+      let confirmButton = warningModal.find('button').at(1);
+      expect(confirmButton.props().disabled).to.be.true;
+      const checkboxes = warningModal.find('input');
+      checkboxes.at(0).simulate('change', {target: {checked: true}});
+      checkboxes.at(1).simulate('change', {target: {checked: true}});
+      confirmButton = body.find('button').at(1);
+      expect(confirmButton.props().disabled).to.be.false;
+    });
   });
 
-  it('warning message requires both checkboxes to be checked to go forward', () => {
-    const body = shallow(
-      <AnimationUploadButton
-        onUploadClick={emptyFunction}
-        shouldRestrictAnimationUpload={true}
-        isBackgroundsTab={false}
-        refreshInRestrictedShareMode={emptyFunction}
-        inRestrictedShareMode={false}
-        showingUploadWarning={emptyFunction}
-        exitedUploadWarning={emptyFunction}
-      />
-    );
-    const uploadButton = body.find(AnimationPickerListItem).at(0);
-    uploadButton.simulate('click');
-    const warningModal = body.find(BaseDialog).at(0);
-    const buttons = warningModal.find('button');
-    console.log(`found ${buttons.length} buttons`);
```

Comment:
@molly-moen removed these console logs as I think they're just leftover from your initial testing, LMK if not

Reply:
yep, good catch!